contracts/gauges/gauge/src/state.rs

```diff
@@ -86,6 +88,19 @@ impl Gauge {
             .map(|r| r.last == Some(r.next))
             .unwrap_or_default()
     }
+    pub fn will_reach_epoch_limit(&self) -> bool {
+        if let Some(total) = self.total_epoch {
+            total == self.count.unwrap_or_default()
+        } else {
+            false
+        }
+    }
+    pub fn increment_gauge_count(&self) -> StdResult<Option<u64>> {
+        Ok(self.count.map_or(Some(0), |o| Some(o + 1)))
+    }
+    pub fn gauge_epoch(&self) -> StdResult<u64> {
+        Ok(self.count.map_or(Some(0), Some).unwrap_or_default())
+    }
```

Comment:
Doc comments for these would be great.
```suggestion

    pub fn will_reach_epoch_limit(&self) -> bool {
        if let Some(total) = self.total_epoch {
            total == self.count.unwrap_or_default()
        } else {
            false
        }
    }
    
    pub fn increment_gauge_count(&self) -> StdResult<Option<u64>> {
        Ok(self.count.map_or(Some(0), |o| Some(o + 1)))
    }
    
    pub fn gauge_epoch(&self) -> StdResult<u64> {
        Ok(self.count.map_or(Some(0), Some).unwrap_or_default())
    }
```

Reply:
And spacing. : )